datalake/src/main/java/com/sequenceiq/datalake/service/sdx/dr/SdxDrService.java

```diff
@@ -0,0 +1,273 @@
+        LOGGER.info("Response from cloudbreak: {}", JsonUtil.writeValueAsString(stackV4Response));
+        ClusterV4Response cluster = stackV4Response.getCluster();
+        if (stackAndClusterAvailable(stackV4Response, cluster)) {
+            return sdxDrSucceeded(stackV4Response);
+        } else {
+            if (Status.BACKUP_FAILED.equals(stackV4Response.getStatus()) ||
+                    Status.RESTORE_FAILED.equals(stackV4Response.getStatus())) {
+                LOGGER.info("{} failed for Stack {} with status {}", pollingMessage, stackV4Response.getName(), stackV4Response.getStatus());
+                return sdxDrFailed(sdxCluster, stackV4Response.getStatusReason(), pollingMessage);
+            } else if (Status.BACKUP_FAILED.equals(stackV4Response.getCluster().getStatus()) ||
+                    Status.RESTORE_FAILED.equals(stackV4Response.getCluster().getStatus())) {
+                LOGGER.info("{} failed for Cluster {} status {}", pollingMessage, stackV4Response.getCluster().getName(),
+                        stackV4Response.getCluster().getStatus());
+                return sdxDrFailed(sdxCluster, stackV4Response.getCluster().getStatusReason(), pollingMessage);
+            } else {
+                if (FINISHED.equals(flowState)) {
+                    LOGGER.info("Flow finished, but Backup/Restore is not complete: {}", sdxCluster.getClusterName());
+                    return sdxDrFailed(sdxCluster, "stack is in improper state", pollingMessage);
+                } else {
+                    return AttemptResults.justContinue();
+                }
+            }
+        }
+    }
+
+    private AttemptResult<StackV4Response> sdxDrFailed(SdxCluster sdxCluster, String statusReason, String pollingMessage) {
+        LOGGER.info("{} failed, statusReason: {}", pollingMessage, statusReason);
+        return AttemptResults.breakFor("SDX " + pollingMessage + " failed '" + sdxCluster.getClusterName() + "', " + statusReason);
+    }
+
+    private AttemptResult<StackV4Response> sdxDrSucceeded(StackV4Response stackV4Response) {
+        LOGGER.info("Database DR operation doe SDX cluster {} is successfull", stackV4Response.getCluster().getName());
+        return AttemptResults.finishWith(stackV4Response);
+    }
+
+    private boolean stackAndClusterAvailable(StackV4Response stackV4Response, ClusterV4Response cluster) {
+        return stackV4Response.getStatus().isAvailable()
+                && cluster != null
+                && cluster.getStatus() != null
+                && cluster.getStatus().isAvailable();
+    }
+
+    /**
+     * Updates the status of the database backup/restore operation.
+     * @param operationId Operation Id
+     * @param status Status of the operation
+     * @param failedReason Failure reason, if any.
+     */
+    public void updateDatabaseStatusEntry(String operationId, SdxDatabaseDrStatus.Status status, String failedReason) {
+        if (Strings.isNullOrEmpty(operationId)) {
+            return;
```

Comment:
we might log this as it should not happen

Reply:
will do.